### PR TITLE
feat: [STO-4451]: STO Pipelines and Deployments pages

### PR DESCRIPTION
**Pull request by chris-harness**
##### Summary:

- Add STO-specific Security Tests page (equivalent to Deployments or Builds
- Add STO-specific Pipelines page
- Use latest `design-system` release to fix the wrong `--purple-500` color, and change existing uses of it to `--purple-600` to maintain the current appearance in browser

##### Jira Links:

https://harness.atlassian.net/browse/STO-4451

##### Screenshots:

Screen Shot 2022-05-16 at 2 08 45 PM

Screen Shot 2022-05-16 at 2 14 06 PM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by chris-harness** [comment]
retrigger jest

**Comment by chris-harness** [comment]
retrigger jest

**Comment by chris-harness** [comment]
Jest failure is intermittent, and passes on my local

**Comment by chris-harness** [comment]
retrigger jest

**Comment by chris-harness** [comment]
retrigger coverage

File: src/modules/70-pipeline/pages/pipeline-deployment-list/PipelineDeploymentList.tsx [hunk 1]
```diff
@@ -153,11 +155,43 @@ function NoDeployments(props: {
...
     isPipelineInvalid
   } = props || {}
+
+  let icon: IconName
+  let img: JSX.Element
+  let noDeploymentString: keyof StringsMap
+  let aboutDeploymentString: keyof StringsMap
+
+  switch (module) {
+    case 'ci':
```
Comment (nanaxuHarness) [inline src/modules/70-pipeline/pages/pipeline-deployment-list/PipelineDeploymentList.tsx:165]: can we use Module enum here

File: src/modules/85-sto-steps/components/STOExecutionCardSummary/STOExecutionCardSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  // Render the message
+  if (message) {
+    return <div className={css.main}>{message}</div>
+  }
+
+  // Otherwise, render the Security Issue counts
+  return (
+    <div className={css.main}>
+      {issueCounts?.critical ? <SeverityPill severity={SeverityCode.Critical} value={issueCounts.critical} /> : <></>}
```
Comment (nanaxuHarness) [inline src/modules/85-sto-steps/components/STOExecutionCardSummary/STOExecutionCardSummary.tsx:68]: can we do `issueCounts?.critical && <SeverityPill severity={SeverityCode.Critical} value={issueCounts.critical} />` here

File: src/modules/85-sto-steps/components/STOExecutionCardSummary/__tests__/STOExecutionCardSummary.test.tsx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+import type { PipelineExecutionSummary } from 'services/pipeline-ng'
+import * as stoService from 'services/sto'
+import type { SeverityPillProps } from '@sto-steps/components/SeverityPill/SeverityPill'
+import STOExecutionCardSummary from '@sto-steps/components/STOExecutionCardSummary/STOExecutionCardSummary'
+
+jest.mock('@sto-steps/components/SeverityPill/SeverityPill', () => ({ severity, value }: SeverityPillProps) => (
+  <div data-testid={severity} data-value={value} />
+))
+
+jest.mock('@blueprintjs/core', () => {
```
Comment (nanaxuHarness) [inline src/modules/85-sto-steps/components/STOExecutionCardSummary/__tests__/STOExecutionCardSummary.test.tsx:23]: why do we need to mock this?

File: src/modules/85-sto-steps/components/STOSideNav/STOSideNav.tsx [hunk 4]
```diff
@@ -71,6 +71,12 @@ const TabPanel: React.FC<TabPanelProps> = ({ isProjectMode }) => {
   // Telemetry?
 
   const showLinks = !isProjectMode || projectIdentifier
+  const params: ProjectPathProps & ModulePathParams & { accountId: string } = {
+    accountId,
+    projectIdentifier,
+    orgIdentifier,
+    module: 'sto'
```
Comment (nanaxuHarness) [inline src/modules/85-sto-steps/components/STOSideNav/STOSideNav.tsx:78]: can we use Module enum here

File: src/modules/85-sto-steps/strings/strings.en.yaml [hunk 5]
```diff
@@ -5,3 +5,11 @@ securityStage:
   description: Scan code, artifacts, and infrastructure for security vulnerabilities.
 targets:
   testTargets: Test Targets
+noScansText: There are no Security Test runs in your project
+noScansRunPipelineText: No Security Test pipelines have run, yet. Click the button below to run a pipeline.
```
Comment (nanaxuHarness) [inline src/modules/85-sto-steps/strings/strings.en.yaml:9]: do we need the `,` here?

File: src/modules/70-pipeline/pages/pipelines/PipelinesPage.tsx [hunk 6]
```diff
@@ -283,7 +283,7 @@ function PipelinesPage({ mockData }: CDPipelinesPageProps): React.ReactElement {
   const defaultQueryParamsForPiplines = {
     accountIdentifier: accountId,
     projectIdentifier,
-    module,
+    module: module,
```
Comment (tan-nhu) [inline src/modules/70-pipeline/pages/pipelines/PipelinesPage.tsx:286]: Value and key is the same. We can ignore value

File: src/modules/70-pipeline/pages/pipeline-deployment-list/PipelineDeploymentList.tsx (hunk 1 above)
Comment (chris-harness) [reply src/modules/70-pipeline/pages/pipeline-deployment-list/PipelineDeploymentList.tsx:165]: Refactored to use the `Module` type for this prop

File: src/modules/85-sto-steps/components/STOExecutionCardSummary/STOExecutionCardSummary.tsx (hunk 2 above)
Comment (chris-harness) [reply src/modules/85-sto-steps/components/STOExecutionCardSummary/STOExecutionCardSummary.tsx:68]: Yes! fixed

File: src/modules/70-pipeline/pages/pipeline-deployment-list/images/security-tests-illustration.svg [hunk 7]
```diff
@@ -0,0 +1,74 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="120" height="75" fill="none" viewBox="0 0 120 75">
```
[the remark this answers is not in the document]
Comment (chris-harness) [reply src/modules/70-pipeline/pages/pipeline-deployment-list/images/security-tests-illustration.svg:1]: Not sure it's necessary. It renders the expected, consistent size due to existing CSS.

File: src/modules/70-pipeline/pages/pipelines/PipelinesPage.tsx (hunk 6 above)
Comment (chris-harness) [reply src/modules/70-pipeline/pages/pipelines/PipelinesPage.tsx:286]: Yeah, noticed this as well. Fixed

File: src/modules/85-sto-steps/components/STOExecutionCardSummary/__tests__/STOExecutionCardSummary.test.tsx (hunk 3 above)
Comment (chris-harness) [reply src/modules/85-sto-steps/components/STOExecutionCardSummary/__tests__/STOExecutionCardSummary.test.tsx:23]: It's done to inject a test ID

File: src/modules/85-sto-steps/components/STOSideNav/STOSideNav.tsx (hunk 4 above)
Comment (chris-harness) [reply src/modules/85-sto-steps/components/STOSideNav/STOSideNav.tsx:78]: The `Module` type ensures it is valid. Not aware of any enum for this.

File: src/modules/85-sto-steps/strings/strings.en.yaml (hunk 5 above)
Comment (chris-harness) [reply src/modules/85-sto-steps/strings/strings.en.yaml:9]: removed

**Comment by chris-harness** [comment]
retrigger coverage

**Comment by chris-harness** [comment]
retrigger jest

**Comment by chris-harness** [comment]
retrigger coverage

**Comment by chris-harness** [comment]
retrigger jest

**Comment by chris-harness** [comment]
retrigger eslint

File: src/modules/70-pipeline/pages/deployments-list/__tests__/DeploymentsList.test.tsx [hunk 8]
```diff
@@ -117,4 +117,20 @@ describe('Pipeline Deployments List', () => {
     expect(queryByText('pipeline.noBuildsText')).toBeTruthy()
     expect(queryAllByText('buildsText')).toBeTruthy()
   })
+
+  test('STO module', () => {
+    // eslint-disable-next-line
+    // @ts-ignore
+    useParams.mockImplementation(() => {
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/pages/deployments-list/__tests__/DeploymentsList.test.tsx:124]: this should be passed as part of the test wrapper. The test wrapper has a prop for pathParams

File: src/modules/10-common/utils/routeUtils.ts [hunk 9]
```diff
@@ -219,3 +221,13 @@ export const validateReturnUrl = (url: string): boolean => {
 }
 
 export const returnLaunchUrl = (url: string): string => `${window.location.pathname.replace(/\/ng\//, '/')}${url}`
+
+export const moduleFromPathParam = (module: Module): ModuleName | null => {
```
Comment (AshwinBhatkal-Harness) [inline src/modules/10-common/utils/routeUtils.ts:225]: What do we need this for?
We already have moduleToModuleNameMapping, can we use that instead?

File: src/modules/70-pipeline/pages/pipeline-deployment-list/PipelineDeploymentList.tsx [hunk 10]
```diff
@@ -241,10 +283,10 @@ export default function PipelineDeploymentList(props: PipelineDeploymentListProp
     status
   })
 
-  const isCIModule = module === 'ci'
   const { getString } = useStrings()
   const hasFilterIdentifier = getHasFilterIdentifier(filterIdentifier)
 
+  const module: Module = moduleFromPathParam(modulePathParam) || 'cd'
```
Comment (vkbansal) [inline src/modules/70-pipeline/pages/pipeline-deployment-list/PipelineDeploymentList.tsx:289]: This is not the right way of doing this. We already have a `useModuleInfo` hook for this kind of use case. Please use that.

File: src/modules/10-common/utils/routeUtils.ts (hunk 9 above)
Comment (vkbansal) [inline src/modules/10-common/utils/routeUtils.ts:225]: Use `useModuleInfo` hook instead. This is not reliable.

**Review by vkbansal — CHANGES_REQUESTED** [review]

File: src/modules/70-pipeline/pages/pipeline-deployment-list/PipelineDeploymentList.tsx (hunk 10 above)
Comment (chris-harness) [reply src/modules/70-pipeline/pages/pipeline-deployment-list/PipelineDeploymentList.tsx:289]: Done

File: src/modules/10-common/utils/routeUtils.ts (hunk 9 above)
Comment (chris-harness) [reply src/modules/10-common/utils/routeUtils.ts:225]: Done

**Comment by chris-harness** [comment]
retrigger jest

File: src/modules/70-pipeline/pages/deployments-list/__tests__/DeploymentsList.test.tsx (hunk 8 above)
Comment (chris-harness) [reply src/modules/70-pipeline/pages/deployments-list/__tests__/DeploymentsList.test.tsx]: fixed tests to do it the right way

**Review by nanaxuHarness — DISMISSED** [review]

**Review by tan-nhu — APPROVED** [review]

**Review by tan-nhu — APPROVED** [review]

**Review by vasuki-kalluri — APPROVED** [review]

**Review by vkbansal — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Pull request merged**